### PR TITLE
Support availability

### PR DESCRIPTION
**Pull request by ivarshav**
Support availability of an event.

image

### CODE REVIEW

File: app/routers/event.py [hunk 1]
```diff
@@ -215,6 +222,7 @@ def create_event(db: Session, title: str, start, end, owner_id: int,
         location=location,
         invitees=invitees_concatenated,
         category_id=category_id,
+        availability=availability
```
Comment (yammesicka) [inline app/routers/event.py:225]: Add a trailing comma :)

File: app/templates/agenda.html [hunk 2]
```diff
@@ -44,7 +44,11 @@ <h1>{{ start_date.strftime("%d/%m/%Y") }} - {{end_date.strftime("%d/%m/%Y") }}</
     {% for events_date, events_list in events.items() %}
     <div class="shown_dates">{{ events_date.strftime("%d/%m/%Y") }}</div>
         {% for event in events_list %}
-        <div class="event_line"> {{ event[0].start.time().strftime("%H:%M") }} - <a href="/event/view/{{ event[0].id }}">{{ event[0].title }}</a> , duration: {{ event[1] }}</div>
+        <div class="event_line">
+            {% set availability = 'Busy' if event[0].availability == True else 'Free' %}
+            <div class="{{ availability | lower }}" title="{{ availability }}"></div>
+            <div>{{ event[0].start.time().strftime("%H:%M") }} - <a href="/event/view/{{ event[0].id }}">{{ event[0].title }}</a> , duration: {{ event[1] }}</div>
```
Comment (yammesicka) [inline app/templates/agenda.html:50]: Escape the `event[0].title` (and everything else coming from the user) with `| safe`

File: app/templates/agenda.html [hunk 3]
```diff
@@ -44,7 +44,11 @@ <h1>{{ start_date.strftime("%d/%m/%Y") }} - {{end_date.strftime("%d/%m/%Y") }}</
     {% for events_date, events_list in events.items() %}
     <div class="shown_dates">{{ events_date.strftime("%d/%m/%Y") }}</div>
         {% for event in events_list %}
-        <div class="event_line"> {{ event[0].start.time().strftime("%H:%M") }} - <a href="/event/view/{{ event[0].id }}">{{ event[0].title }}</a> , duration: {{ event[1] }}</div>
+        <div class="event_line">
+            {% set availability = 'Busy' if event[0].availability == True else 'Free' %}
```
Comment (yammesicka) [inline app/templates/agenda.html:48]: Can we be sure there is at least 1 element in `event`?

**Review by yammesicka — CHANGES_REQUESTED** [review]
Awesome job! I'd like to suggest some minor improvements :)

File: app/templates/agenda.html (hunk 3 above)
Comment (ivarshav) [reply app/templates/agenda.html:48]: AFAIU, the BE sends a list of [(event, event_duration)].

IMO we can simplify this and pass only list of events, will do in other feature.
Thanks!

Comment (yammesicka) [reply app/templates/agenda.html:48]: Sure NP

**Comment by yammesicka** [comment]
Ready to merge from my side. Just please solve the conflicts :)

**Comment by ivarshav** [comment]
Conflicts were solved, will be happy to merge the code, thanks!

File: app/templates/agenda.html (hunk 3 above)
Comment (ivarshav) [reply app/templates/agenda.html:48]: @yammesicka, 
In https://github.com/PythonFreeCourse/calendar/pull/298 the structure is simplified , thanks.

**Pull request merged**